docs/40_smart-contract-guides/80_linking-custom-permission.md

```diff
@@ -2,72 +2,479 @@
+
+## Why use a Custom Permission
+
+Use custom permssions to improve the security on your account by setting a custom permission for calling specific transactions. This will limit the need to use the `owner` and `active` keys and limit the risk of these permissions being compromised.
+
+[[info]]
+| Custom permissions always have a parent permission. The parent permission has the authority to execute the same transactions as the custom permission, and to recover the custom permission keys if that custom permission is compromised.
+
+## Using Custom Permissions
+
+The following examples show you how to:
+
+* Create a Custom Permission
+* Link a Custom Permission
+* Unlink a Custom Permission
+* Delete a Custom Permission
+
+In this example you will use:
+
+* An account called _bob_, and the keys to control this account stored in a local wallet.
+* An account called _scholder_, and the keys to control this account stored in a local wallet.
+* A smart contract called _hello_ which has been deployed to the _scholder_ account. This smart contract has three actions:
+	- `what(eosio::name user)`
+	- `why(eosio::name user)`
+	- `how(eosio::name user)`
+
+In this example you will create two custom permissions on the _bob_ account:
+* A custom permission, _customp1_,  whose parent is active
+* A custom permission, _customp2_, whose parent is _customp1_
+
+You will link the custom permissions to smart contract actions:
+* Link _customp1_ to `what(eosio::name user)`
+* Link _customp2_ to `how(eosio::name user)`
+
+You will see that the `active` permission can call all the actions.
+You will see that the _customp1_  permission can call the `why` and `how` actions.
+You will see that the _customp2_  permission can only call the `how` action.
+
+### The simple smart contract
+
+The simple smart contract code used in the examples:
+
+```c++
+
+#include <eosio/eosio.hpp>
+class [[eosio::contract]] hello : public eosio::contract {
+  public:
+      using eosio::contract::contract;
+      [[eosio::action]] void what( eosio::name user ) {
+         print( "hi, what do you want ", user);
+      }
+
+      [[eosio::action]] void why( eosio::name user ) {
+         print( "why not ", user);
+      }
+
+      [[eosio::action]] void how( eosio::name user ) {
+         print( "how are you ", user);
+      }
+};
+
+```
+
+Deploy this smart contract to your running blockchain.
+
+[[info]]
+| This simple sample  smart contract does not do any authorization checking, i.e. does not use  `require_auth`. This tutorial demonstrates **native authorization checking**.
```

Comment:
can you please link the [authorization checking](https://developers.eos.io/manuals/eosio.cdt/latest/best-practices/securing_your_contract/#1-authorization-checks) text to this doc page https://developers.eos.io/manuals/eosio.cdt/latest/best-practices/securing_your_contract/#1-authorization-checks

Reply:
added link